R/pin-read-write.R

```diff
@@ -71,15 +71,19 @@ pin_read <- function(board, name, version = NULL, hash = NULL, ...) {
 #' @export
 pin_write <- function(board, x,
                       name = NULL,
+                      ...,
                       type = NULL,
                       title = NULL,
                       description = NULL,
                       metadata = NULL,
                       versioned = NULL,
                       tags = NULL,
-                      ...,
                       force_identical_write = FALSE) {
   check_board(board, "pin_write", "pin")
+  dots <- list2(...)
+  if (!is_empty(dots) && is.null(names(dots[1]))) {
```

Comment:
I don't think the logic is quite right here as `names` will be `NULL` if all the elements are unnamed, but if there's one named element, it'll be `""`.

I don't know if `...names()` is available in old enough R for us to use it, but you could also try something like this:

```R
if (!missing(...) && !(is.null(...names())) || ...names()[[1]] == ""))
```

Reply:
Looks like we do not yet have `...names()` in oldrel-3 and oldrel-4.